channeld/channeld.c

```diff
@@ -202,6 +207,52 @@ static void billboard_update(const struct peer *peer)
 		       announce_status, shutdown_status);
 }
 
+/* used in billboard_update to say how many blocks to go! */
+static char *get_funding_depth(const struct peer *peer)
+{
+	status_trace("Please wait. We are trying to get depth...");
+	u8 *msg, *reply;
+	u32 minimum_depth, funding_depth;
+	char *infor = tal_arr(tmpctx, char, 0);
+	struct bitcoin_txid txid;
+
+	msg = towire_channel_funding_depth(tmpctx, &peer->channel->funding_txid);
+
+	/* wait for the reply, this loop may cost some minutes */
+	reply = master_wait_sync_reply(tmpctx,
+				       cast_const(struct peer *, peer),
+				       msg,
+				       WIRE_CHANNEL_FUNDING_DEPTH_REPLY);
+
+	if (!fromwire_channel_funding_depth_reply(reply,
+						  &txid,
+						  &minimum_depth,
+						  &funding_depth))
+		master_badmsg(WIRE_CHANNEL_FUNDING_DEPTH_REPLY, msg);
+
+	/* the struct channel in channeld.c and the struct channel in
+	** channel_control.c are different, there should compare txid
+	** to confirm channeld and master are saying the same
+	** (funding_txid)channel */
+	if(!bitcoin_txid_eq(&peer->channel->funding_txid, &txid))
+		status_failed(STATUS_FAIL_INTERNAL_ERROR,
+				"Could not get funding depth: %s",
+				type_to_string(tmpctx, struct bitcoin_txid,
+				&peer->channel->funding_txid));
+	u32 diff = (minimum_depth > funding_depth)? minimum_depth-funding_depth: 0;
+	infor = tal_fmt(tmpctx,
+			"funding %s depth %u, minimum depth %u,"
+			" need %u blocks(about %u min) to go!",
+			type_to_string(tmpctx, struct bitcoin_txid,
+				       &peer->channel->funding_txid),
+			funding_depth,
+			minimum_depth,
+			diff,
+			10 * diff);
```

Comment:
Yes, it would be incompatible with with litecoin. 

In general, while this is well intentioned and informative, it's probably best to leave out a time estimate as the actual variation in block time is considerable. A 30 minute wait for a confirmation is quite ordinary, and it may mislead users into thinking a block _will_ be mined shortly, when it's not deterministic.

Reply:
thank you all, It's better to delete this time information.